### ISSUE TITLE
[v2-unstable] Since last commit, --help fails

### ISSUE THREAD

**Issue by abourget** [alecthomas/kingpin#31]
The interspearsed arguments are pretty cool, but since the last commit you can't call with "--help" !

```
...
  download [<flags>] <projectRev>
    Retrieve a build Artifact
```

then:
```
$ alchemist download --help
alchemist: error: required argument 'projectRev' not provided, try --help
```

**Comment by alecthomas** [comment]
Yes, I'm in the middle of rewriting the help system.

Basically you'll be able to put `--help` at any point and it will look at the fully parsed context to determine what to display. Previously it was pretty hacky.

**Comment by abourget** [comment]
ok, will you update this when it's fixed ? :)

**Comment by alecthomas** [comment]
Definitely!

**Comment by abourget** [comment]
Any news on that front ?

**Comment by alecthomas** [comment]
Sorry, my day job has intruded somewhat. I'll try to get to it soon.

**Comment by abourget** [comment]
thanks :) I appreciate!

**Comment by alecthomas** [comment]
This was fixed in 7b8f686015f25d92a5db8d01dc15455b53da9c9d.
